pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -422,14 +422,15 @@ func (r *ReconcileScanSettingBinding) verifyRoleInScanSettingBinding(
 		prfRef := &binding.Profiles[i]
 
 		key := types.NamespacedName{Namespace: binding.Namespace, Name: prfRef.Name}
-		tp, err := getUnstructured(r, binding, key, prfRef.Kind, prfRef.APIGroup, logger)
+		prfObj, err := getUnstructured(r, binding, key, prfRef.Kind, prfRef.APIGroup, logger)
 		if err != nil {
-			return fmt.Errorf("error getting TailoredProfile %s: %w", prfRef.Name, err)
+			return fmt.Errorf("error getting Profile %s: %w", prfRef.Name, err)
 		}
 
-		profileType, err := getScanType(tp.GetAnnotations())
+		profileType, err := getTpScanType(r, binding, prfRef.APIGroup, prfObj, logger)
 		if err != nil {
-			return fmt.Errorf("error getting profile type %s: %w", prfRef.Name, err)
+			logger.Info("error getting scan type, assuming Platform", "profile", prfRef.Name, "error", err)
+			profileType = compliancev1alpha1.ScanTypePlatform
```

Comment:
If this case happens and we use the platform scan type, but we couldn't get the Tailored Profile (on line 701), will this cascade into another failure down the line?

Asking because this doesn't seem like a situation we can recover from.


 and we return, we're signaling an error happened *and* that the caller should use the platform scan.

If we can't fetch the Tailored Profile right now, will

Reply:
I think most of the errors (like plain `Get`s) should be retried, I marked one as non-retriable. For easier review, here is the diff:
```
diff --git a/pkg/controller/scansettingbinding/scansettingbinding_controller.go b/pkg/controller/scansettingbinding/s>
index 7d5ad113..a9a1a292 100644
--- a/pkg/controller/scansettingbinding/scansettingbinding_controller.go
+++ b/pkg/controller/scansettingbinding/scansettingbinding_controller.go
@@ -693,6 +693,8 @@ func getTpScanType(
        if err == nil {
                return profileType, nil
        } else if prfObj.GetKind() != "TailoredProfile" {
+               // profiles are pretty much always annotated. If not, let's just assume it's a platform scan. If not,
+               // the scan would just fail later..
                logger.Info("error getting profile scan type, assuming Platform", "profile", prfObj.GetName(), "error>
                return compliancev1alpha1.ScanTypePlatform, nil
        }
@@ -704,6 +706,9 @@ func getTpScanType(
                        "TailoredProfile", prfObj.GetName())
                return compliancev1alpha1.ScanTypePlatform, nsErr
        } else if !found {
+               // if there is no extends, then this is a custom-created profile. Custom profiles would indicate that>
+               // are node profiles either by using the -node suffix or by being properly annotated. Otherwise, even
+               // the plain scan profile would assume platform, so let's do the same here.
                logger.Info("No extends field found in tailored profile", "TailoredProfile", prfObj.GetName())
                return compliancev1alpha1.ScanTypePlatform, nil
        }
@@ -716,7 +721,7 @@ func getTpScanType(
 
        extendsType, err := getScanType(extendsObj.GetAnnotations())
        if err != nil {
-               return extendsType, err
+               return extendsType, common.NewNonRetriableCtrlError("error getting scan type for profile %s: %w", ext>
        }
 
        logger.Info("Got scan type from parent profile", "tailoredProfile", prfObj.GetName(), "profile", extendsObj.G>
```